src/components/CIPipelineN/Build.tsx

```diff
@@ -64,13 +66,19 @@ export function Build({
         const _formData = { ...formData }
         let isPrevWebhook =
             _formData.ciPipelineSourceTypeOptions.find((sto) => sto.isSelected)?.value === SourceTypeMap.WEBHOOK
+          
         const allMaterials = _formData.materials.map((mat) => {
+            const _type = gitMaterialId === mat.gitMaterialId ? selectedSource.value : mat.type
+            const branchDecision=_type==='SOURCE_TYPE_BRANCH_REGEX'
```

Comment:
Can we rename the variable more specifically? like _type to `sourcetype` and `branchDecision` to isBranchRegexType as it is returning a boolean value after checking